test/main.cpp

```diff
@@ -0,0 +1,27 @@
+#include "ostream"
+
+#define CATCH_CONFIG_MAIN
+#include <catch.hpp>
+#include <mock_external.hpp>
+
+TEST_CASE("mock_external_c, set_send_reaction compiles", ""){
+    auto test_mock = r2d2::communication::mock_external_c();
+    auto frame = r2d2::frame_external_s();
+    std::function<void(
+            const r2d2::frame_external_s&, 
+            std::vector<r2d2::frame_external_s>&
+            )> lam = [&](
+                const r2d2::frame_external_s &fr, 
+                std::vector<r2d2::frame_external_s> &retvec
+                )->void{
```

Comment:
`-> void` as a lambda return type is inferred and can be omitted.

Reply:
However it does not have to be